docs/next/guides/upgrade-and-migration/migrating-from-11.1-to-12.0.md

```diff
@@ -64,7 +64,11 @@ As a result, whenever you called [`updateSettings()`](@/api/core.md#updatesettin
 
 #### After
 
-A plugin gets updated only if you update settings related to that particular plugin.
+A plugin's [`updatePlugin()`](@/api/autocolumnsize.md#updateplugin) method gets triggered only when the object passed to [`updateSettings()`](@/api/core.md#updatesettings) contains at least one of the following:
+- The plugin's [`PLUGIN_KEY`](@/guides/building-and-testing/plugins.md#_2-extend-the-baseplugin) (the plugin's main alias)
+- An entry from the plugin's [`SETTING_KEYS`](@/api/baseplugin.md#setting-keys) (a property that stores all additional settings related to the plugin)
+
+As a result, a plugin gets updated only if you update settings related to that particular plugin.
 
 Only the following plugins still get updated on every [`updateSettings()`](@/api/core.md#updatesettings) call:
```

Comment:
This line needs some connection with the new information presented above. 

I propose to rewrite this paragraph as:

```
Since version 12.0.0, Handsontable updates the built-in plugins on a call to `updateSettings()` only if configuration object contains options relevant to the plugin. Only the following plugins still get updated on every [`updateSettings()`](@/api/core.md#updatesettings) call:
```

Reply:
@warpech If we want to stress the fact that the change affects built-in plugins as well, we could do it in a dedicated sub-section, WDYT? [Commit](https://github.com/handsontable/handsontable/pull/9271/commits/b12cf5c68f696342d94d9de786692eefc203f3d6#diff-2dde93270e7af4691dfb637c6e2dc68cc1ea6cfd00a295aff176e331267e9499R74-R88)

Reply:
I don't think any action is required in the case of internal plugins. This changes modifies they way they react to `updateSettings`, but I don't think there's anything the user can do about it (apart from adding the plugin config to the `updateSettings`'s config object) or would like to do about it anyway.

Reply:
If that was true, then the information about built-in plugins (https://github.com/handsontable/handsontable/blob/2bceb5a9cbc75b04d7cfe0690b5f2c8824a4fcbf/docs/next/guides/upgrade-and-migration/migrating-from-11.1-to-12.0.md?plain=1#L74-L88) should be removed from the migration guide altogether. 

However, the reason why we moved this change to the major release, was because it was potentially breaking:

- https://kb.handsontable.com/pages/viewpage.action?pageId=56459614
- https://github.com/handsontable/handsontable/pull/9025

To clarify - you meant that it is breaking because:
1. (custom) plugins need to make sure they observe the relevant configuration options
2. or because is breaking for apps that for some complex reason relied on the fact that the plugins observe too much and update their state.

If the latter is the case, then we could keep it in the migration guide along with the list of the changed plugins. If the developer used any of the plugins, they would be advised to retest their apps. I think that the problem is unlikely and could be skipped in the migration guide, but maybe I am wrong.

Reply:
@warpech @jansiegel – what's our decision here?

My point of view: the scope of the migration guide got pretty big with the recent keyboard shortcut breaking changes, so, if the built-in plugin scenario is very unlikely, I'd remove that section.

Reply:
@warpech @kirszenbaum 
> To clarify - you meant that it is breaking because:
> 1. (custom) plugins need to make sure they observe the relevant configuration options
> 2. or because is breaking for apps that for some complex reason relied on the fact that the plugins observe too much and update their state.

Basically, both. _Potentially_ some plugins might have relied on a larger set of plugins being updated, but if that was the case, our tests should have found some problems with it, and as far as i know, no problems of that nature emerged, correct me if I'm wrong, @aninde?

Based on that, I'd say that we could skip the information about the internal plugins.

Reply:
OK, so let's remove the "Built-in plugins" subsection.

And add a new sentence after:

> Handsontable 12.0.0 changes how the [updatePlugin()](http://localhost:8000/docs/next/next/guides/upgrade-and-migration/@/api/autocolumnsize.html) method reacts to [updateSettings()](http://localhost:8000/docs/next/api/core/#updatesettings) calls.

The new sentence should say that: This change is significant for the authors of custom plugins.

Reply:
Done: [e4bd130](https://github.com/handsontable/handsontable/pull/9271/commits/e4bd13028b7638506b15e265913bfbdb2a5eb123)